df_udf/README.md

```diff
@@ -0,0 +1,90 @@
+# Scala / Java UDFS implemented using data frame
+
+User Defined Functions (UDFs) are used for a number of reasons in Apache Spark. Much of the time it is to implement 
+logic that is either very difficult or impossible to implement using existing SQL/Dataframe APIs directly. But they
+are also used as a way to standardize processing logic across an organization or for code reused.
+
+But UDFs come with some downsides. The biggest one is visibility into the processing being done. SQL language that
+can be highly optimized. But a UDF in most cases is a black box, that the SQL optimizer cannot do anything about.
```

Comment:
"SQL language that .." is not a complete sentence since a verb is missing. Maybe supposed to be, "SQL is a language that ..."?